db/blob_log_format.cc

```diff
@@ -0,0 +1,245 @@
+#include "db/blob_log_format.h"
+#include "util/coding.h"
+#include "util/crc32c.h"
+
+
+namespace rocksdb {
+namespace blob_log {
+
+
+BlobLogHeader::BlobLogHeader()
+: magic_number_(kMagicNumber),
+  has_ttl_(false),
+  has_ts_(false)
+{
+}
+
+BlobLogFooter::BlobLogFooter()
+: magic_number_(kMagicNumber),
+  blob_count_(0),
+  has_ttl_(false), has_ts_(false)
+{
+}
+
+Status BlobLogFooter::DecodeFrom(Slice* input)
+{
+  Status s;
+  uint32_t offset = 0;
+
+  const char *ptr = input->data();
+  uint32_t mn = DecodeFixed32(ptr);
```

Comment:
Put the magic number as the last field of the file. Otherwise other type of the files can be identified as this type too.